routes/data.js

```diff
@@ -1,8 +1,73 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express');
+const router = express.Router();
+const firebase = require('firebase');
+
+if (!firebase.apps.length) {
+	firebase.initializeApp(config);
+}
+
+// Creates connection to database.
+const db = firebase.database();
+// Links to head of database.
+const ref = db.ref("live_weller");
+// Links to markets list.
+const marketsRef = ref.child("markets");
 
 router.get('/', function(req, res, next) {
 	res.render('data');
 });
 
+router.post('/', function(req, res) {
+	if (req.body.type === "general") {
+		/* Levels chart */
+		let levZero = 0;
+		let levOne = 0;
+		let levTwo = 0;
+		let levThree = 0;
+
+		/* Store Type chart */
+		let small = 0;
+		let medium = 0;
+		let large = 0;
+		let convenience = 0;
+		
+		marketsRef.once('value', function(snapshot) {
+			snapshot.forEach(function(childSnapshot) {
+				const childData = childSnapshot.val();
+				const level = parseInt(childData.marketInfo.marketLevel);
+				const type = childData.marketInfo.storeType;
+
+				if (level == 0) {levZero++;}
+				else if (level == 1) {levOne++;}
+				else if (level == 2) {levTwo++;}
+				else if (level == 3) {levThree++;}
+
+				if (type == "Small") {small++;}
```

Comment:
same thing here with an object where keys correspond to `type` and values correspond to number of markets with that type:
```javascript
let types = [];
for each market snapshot {
    const marketType = childData.marketInfo.storeType.toLowerCase();
    types[marketType]++;
}
```
and a populated `types` would look like
```javascript
{
    "small": 2,
    "medium": 5,
    "large": 1,
    "convenience": 2,
}
```
with whatever values but basically an object with 2 small markets, 5 medium markets, 1 large market, and 2 convenience markets.